src/sidebar/components/Annotation/AnnotationEditor.js

```diff
@@ -43,63 +47,86 @@ function AnnotationEditor({
   );
 
   const store = useStoreProxy();
-  const draft = store.getDraft(annotation);
   const group = store.getGroup(annotation.group);
 
-  if (!draft) {
-    // If there's no draft, we can't be in editing mode
-    return null;
-  }
-
   const shouldShowLicense =
     !draft.isPrivate && group && group.type !== 'private';
 
   const tags = draft.tags;
   const text = draft.text;
   const isEmpty = !text && !tags.length;
 
-  const onEditTags = ({ tags }) => {
-    store.createDraft(draft.annotation, { ...draft, tags });
-  };
+  const onEditTags = useCallback(
+    tags => {
+      store.createDraft(draft.annotation, { ...draft, tags });
+    },
+    [draft, store]
+  );
 
   /**
    * Verify `newTag` has content and is not a duplicate; add the tag
    *
    * @param {string} newTag
    * @return {boolean} - `true` if tag is added
    */
```

Comment:
Something I missed when this code was initially written is that parameter and return type documentation for callbacks passed to hooks needs to be attached to the closure rather than the hook call.

This does not work as expected:

```js
/** @param {SomeType} param */
const someCallback = useCallback(param => { ... });
```

This does work:

```js
const someCallback = useCallback(
  /** @param {SomeType} param */
  param => { ... }
);
```

This also applies to other cases where we wrap functions (for example, to debounce them or memoize the results).